js/src/setup-mc/setup-stepper/setup-free-listings/useAutoSaveShippingRatesEffect.js

```diff
@@ -1,15 +1,37 @@
+/**
+ * External dependencies
+ */
+import { useCallback } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
-import { useAppDispatch } from '.~/data';
 import useIsEqualRefValue from '.~/hooks/useIsEqualRefValue';
 import useDebouncedCallbackEffect from '.~/hooks/useDebouncedCallbackEffect';
+import useSaveShippingRates from '.~/hooks/useSaveShippingRates';
 
 const useAutoSaveShippingRatesEffect = ( shippingRates ) => {
-	const { saveShippingRates } = useAppDispatch();
+	const { saveShippingRates } = useSaveShippingRates();
 	const shippingRatesRefValue = useIsEqualRefValue( shippingRates );
 
-	useDebouncedCallbackEffect( shippingRatesRefValue, saveShippingRates );
+	/**
+	 * A `saveShippingRates` callback that catches error and do nothing.
+	 * We don't want to show error messages for this auto save feature,
+	 * and want it to fail silently in the background.
+	 */
+	const saveShippingRatesCallback = useCallback(
+		async ( value ) => {
+			try {
+				await saveShippingRates( value );
+			} catch ( error ) {}
+		},
+		[ saveShippingRates ]
+	);
+
+	useDebouncedCallbackEffect(
+		shippingRatesRefValue,
+		saveShippingRatesCallback
+	);
```

Comment:
❔ 
Due to the async call of debounce, it requires wrapping an additional layer of function to catch the error. Considering the problem that the debounce processing was trying to solve already doesn't exist, and the [consistency reason](https://github.com/woocommerce/google-listings-and-ads/pull/1267#discussion_r820441914) I didn't understand, it presents a more unnecessary complexity now.

If it doesn't offer any benefit to maintain consistency with the debounce processing of `useAutoSaveSettingsEffect`, I would still recommend simplifying it.

Reply:
I have created issue https://github.com/woocommerce/google-listings-and-ads/issues/1335 to simplify things even further, by replacing the auto effects with Form `onChange`. We can address the debounce processing there, and by having things together inside `onChange` handler (instead of separate auto effects), I think it should make things easier to understand too. 🙂